lua/graph.lua

```diff
@@ -851,12 +850,15 @@ end
 --returns 1 if you should filter this way 0 otherwise
 function filter_tags_generic(kv)
 
-  if (kv["highway"] == "construction" or kv["highway"] == "proposed") then
+  if (kv["highway"] == "construction" and kv["construction"] == nil) or kv["highway"] == "proposed" then
     return 1
   end
 
   --figure out what basic type of road it is
   local forward = highway[kv["highway"]]
+  if kv["highway"] == "construction" then
+    forward = highway[kv["construction"]]
```

Comment:
road class should be defined in `construction` tag https://wiki.openstreetmap.org/wiki/Tag:highway%3Dconstruction